app/forms/declaration/PackageInformation.scala

```diff
@@ -17,50 +17,84 @@
 package forms.declaration
 
 import forms.DeclarationPage
-import play.api.data.Forms.{number, text}
+import models.DeclarationType.{CLEARANCE, DeclarationType}
+import play.api.data.Forms.{number, optional, text}
 import play.api.data.{Form, Forms}
 import play.api.libs.json.{JsValue, Json}
 import services.PackageTypes
 import utils.validators.forms.FieldValidator._
 
-case class PackageInformation(typesOfPackages: String, numberOfPackages: Int, shippingMarks: String) {
+case class PackageInformation(typesOfPackages: Option[String], numberOfPackages: Option[Int], shippingMarks: Option[String]) {
   def toJson: JsValue = Json.toJson(this)(PackageInformation.format)
 
-  def typesOfPackagesText: String = PackageTypes.findByCode(typesOfPackages).asText()
+  def typesOfPackagesText: Option[String] =
+    for {
+      types <- typesOfPackages
+      packageType <- PackageTypes.findByCode(types)
+    } yield packageType.asText()
 }
 
 object PackageInformation extends DeclarationPage {
 
-  def fromJsonString(value: String): Option[PackageInformation] = Json.fromJson(Json.parse(value)).asOpt
+  private def fromValues(typesOfPackages: String, numberOfPackages: Int, shippingMarks: String): PackageInformation =
+    PackageInformation(typesOfPackages = Some(typesOfPackages), numberOfPackages = Some(numberOfPackages), shippingMarks = Some(shippingMarks))
+
+  private def toValues(packageInformation: PackageInformation): Option[(String, Int, String)] =
+    for {
+      typesOfPackages <- packageInformation.typesOfPackages
+      numberOfPackages <- packageInformation.numberOfPackages
+      shippingMarks <- packageInformation.shippingMarks
+    } yield (typesOfPackages, numberOfPackages, shippingMarks)
 
-  def require1Field[T](fs: (T => Option[_])*): T => Boolean =
-    t => fs.exists(f => f(t).nonEmpty)
+  def fromJsonString(value: String): Option[PackageInformation] = Json.fromJson(Json.parse(value)).asOpt
 
   implicit val format = Json.format[PackageInformation]
 
   val formId = "PackageInformation"
   val limit = 99
 
-  val mapping = Forms
+  private val NumberOfPackagesLimitLower = 0
+  private val NumberOfPackagesLimitUpper = 99999
+
+  val mappingAllFieldsMandatory = Forms
     .mapping(
       "typesOfPackages" ->
         text()
           .verifying("supplementary.packageInformation.typesOfPackages.empty", nonEmpty)
           .verifying("supplementary.packageInformation.typesOfPackages.error", isEmpty or isContainedIn(PackageTypes.all.map(_.code))),
       "numberOfPackages" ->
         number()
-          .verifying("supplementary.packageInformation.numberOfPackages.error", q => q > 0 && q <= 999999),
+          .verifying("supplementary.packageInformation.numberOfPackages.error", isInRange(NumberOfPackagesLimitLower, NumberOfPackagesLimitUpper)),
```

Comment:
Personally I'm not feeling that moving this range to the variables make sense. For me it's always more complicated. Preferable `isInRange(0, 99999)`

Reply:
I think contrary, so if someone alse would like to share their opinion, we could decide then.

Reply:
`isInRange(0, 99999)` is easier to parse as a reader.  Guess it depends on whether those values are used elsewhere.     The capitalisation of the constants looks wrong somehow though.

Reply:
This is driven by scala-style guideline: https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods
Specifically: "Constant names should be in upper camel case. Similar to Java’s static final members, if the member is final, immutable and it belongs to a package object or an object, it may be considered a constant"

Reply:
And as you've noticed, these values are used in 2 places.

Reply:
OK.  But when U was a lad and did Java, constants were ALL_CAPS.  You kids with your fancy new ideas!